compiler/rustc_trait_selection/src/traits/dyn_compatibility.rs

```diff
@@ -53,6 +63,17 @@ fn dyn_compatibility_violations(
     debug_assert!(tcx.generics_of(trait_def_id).has_self);
     debug!("dyn_compatibility_violations: {:?}", trait_def_id);
 
+    // Check for `AsyncFn` traits first to avoid reporting various other
+    // errors about the details of why these traits aren't object-safe.
+    for supertrait in tcx.supertrait_def_ids(trait_def_id) {
+        if tcx.trait_is_async_fn(supertrait) {
+            let fn_trait = tcx.item_name(supertrait);
```

Comment:
defer this call to `item_name` to the error reporting.

Reply:
Oh, lol, you can't. If you want, you could probably remove the `PartialOrd` implementation from `DynCompatibilityViolation`, and change the one(?) sort call in error reporting to stop doing that.

Reply:
Yeah, I did try that initially! Hence the comments about my annoyance that I couldn't put a `DefId` in the `DynCompatibilityViolation`. I don't think I want to extend this PR to removing the `PartialOrd` impl + sorting, as I anticipate that'll affect a number of other error outputs as well.